src/FunctionApps/python/IntakePipeline/conversion.py

```diff
@@ -1,18 +1,62 @@
 import logging
 import re
 import requests
+import hl7
 from typing import List, Dict
 
 
-def clean_message(message: str, delimiter: str = "\n") -> str:
-    cleaned_message = re.sub("[\r\n]+", delimiter, message)
+def clean_message(message: str) -> str:
+    parsed_message = None
+    try:
+        parsed_message = hl7.parse(message)
+
+        # MSH-7
+        cleansed_datetime = normalize_hl7_datetime(parsed_message[0][7][0])
+        parsed_message[0][7][0] = cleansed_datetime
+
+    except Exception:
+        logging.exception(
+            "Exception occurred while cleaning message.  "
+            + "Passing through original message."
+        )
+
+        return message
+
+    return str(parsed_message)
+
+
+def normalize_hl7_datetime(hl7_datetime: str) -> str:
+    hl7_datetime_parts = re.match(
+        r"(\d+)(\.)?(\d+)?([+-])?(\d+)?", hl7_datetime
+    ).groups()
+
+    # Date Base
+    normalized_datetime = hl7_datetime_parts[0][:14]  # First 14 digits
```

Comment:
This also makes me nervous. Should we be converting incoming datetime strings into some kind of known format using Python functions or the `datetime` modules before we chunk the first 14 digits? I don't know why programmers still haven't figured out format standardization around dates but I feel like there's literally always some format spec error.

Reply:
I'm a little leery of the RegEx here as well, but I'm happy to be convinced it is sufficiently robust through testing. Definitely a simple solution.

Reply:
I've been doing some extensive testing and it seems to meet our needs.  It is admittedly narrowly defined but works for this use case.

Reply:
Works for me. I'm okay to move forward with this and keep an eye on things as more data moves through the pipeline. Also totally appreciate the initial intent was to get this done for demo.

Reply:
I agree, I'm thoroughly satisfied with the amount of testing that went into this. I'm perfectly convinced that we can use these regexes here until proven otherwise.